pkg/controllers/provisioningv2/rke2/rkecontrolplane/controlplane.go

```diff
@@ -122,7 +122,7 @@ func (h *handler) doRemove(cp *rkev1.RKEControlPlane) func() (string, error) {
 
 		for _, machine := range allMachines {
 			// Only delete custom machines. Custom machines can be added outside the UI, so it is important to check each machine.
-			if machine.APIVersion != "rke-machine.cattle.io" || machine.Kind != "CustomMachine" {
+			if machine.Spec.InfrastructureRef.APIVersion != "rke.cattle.io/v1" || machine.Spec.InfrastructureRef.Kind != "CustomMachine" {
```

Comment:
Are we 100% guaranteed that `Spec` / `InfrastructureRef` are non-nil?

Reply:
Honestly, we don't even need to delete the controlplane nodes in a custom setup since they are owned by the CAPI cluster instead of the RKEControlPlane object. If those fields are empty (they aren't pointers so they can never be nil) the CAPI machine object will get cleaned up by CAPI.

Reply:
Oops, my bad. I looked into wrong type from `tests` package instead of CAPI upstream ones.